pyvista/plotting/colors.py

```diff
@@ -875,6 +882,25 @@ def __hash__(self):  # pragma: no cover
         """Hash calculation."""
         return hash((self._red, self._green, self._blue, self._opacity))
 
+    def __getitem__(self, item):
+        """Support indexing the float RGBA representation for backward compatibility."""
+        if not isinstance(item, (str, int, np.integer)):
+            raise TypeError("Invalid index specified, only strings and integers are supported.")
+        if isinstance(item, str):
+            for i, cnames in enumerate(self.CHANNEL_NAMES):
+                if item in cnames:
+                    item = i
+                    break
+            else:
+                raise ValueError(f"Invalid string index {item!r}.")
+        if not 0 <= item < 4:
```

Comment:
I just realised this doesn't support `color[-1]` for alpha... Do we want to touch this? :laughing: Since negative indices can be ambiguous due to the 3-vs-4 length distinction depending on the alpha channel, this might not be a significant issue. But it's better to get everything right.

Reply:
Hmm, can we just accept any integer or slice and pass it to the `float_rgba` for indexing?

Reply:
And let that raise any exceptions for wrong indices/slices

Reply:
Seems to work fine (see f4fee55)

Reply:
I would've handled the exception on our own, but since this is for backwards compat and we've handled string indices, you're right that we can just let the tuple raise when it has to. And if anyone in the future raises an issue about yet another kind of index, we'll just handle strings and pass everything else to the tuple :smile: